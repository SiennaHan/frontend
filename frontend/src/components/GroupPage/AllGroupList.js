import React from 'react';
import { connect } from 'react-redux';

import { toJoinGroup, gotoGroupDetail } from '../../actions/index.js';

const AllGroup = ({ group, onClickGroup, onClickJoinGroup }) => {
    return (
        <div>
            <div onClick={onClickGroup}>
                {group.id} {group.group_type} {group.group_name}
            </div>
            <button onClick={onClickJoinGroup}>
                JOIN GROUP
            </button>
        </div>
    )
}

class AllGroupList extends React.Component {
    render() {
        return (
            <div>
                <h1>All Group List</h1>
                <ul>
                    {this.props.all_groups.map(group => 
                        <AllGroup 
                            key={group.group_name}
                            group={group}
                            onClickGroup={() => this.props.onToGroupDetail(group.id)}
                            onClickJoinGroup={() => this.props.onJoinGroup(group.id)}
                        />
                    )}  
                </ul>
            </div>
         )
    } 
}

const mapStateToProps = (state) => ({
    all_groups: state.all_groups,
})

const mapDispatchToProps = (dispatch) => ({
    onJoinGroup: (groupid) => toJoinGroup(groupid),
    onToGroupDetail: (groupid) => gotoGroupDetail(groupid)
})

export default connect (mapStateToProps, mapDispatchToProps)(AllGroupList)