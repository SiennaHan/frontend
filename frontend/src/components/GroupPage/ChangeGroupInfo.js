import React from 'react';

const ChangeGroupInfo = ({ group, onClickChangeSubmit }) => {
    let grouptype, groupname;

    const type_options = [
		'Select an Option',
		'Club',
		'Department',
		'Else'
        ]

    const onSubmit = () => {
        console.log("changing group information")
        if(grouptype !== undefined && grouptype.value !== 'Select an Option' && groupname !== undefined) {
            onClickChangeSubmit(group.id, grouptype.value, groupname.value)
        }
    }

    return (
        <div>
            <form onSubmit={e => {
                e.preventDefault()
                onSubmit()
            }}>
                <div>
                    <label>
                    GROUP TYPE
                    </label>
                    <select
                        name="grouptype"
                        ref={ node => {grouptype=node;} }
                        className="type-select"
                    >
                        {type_options.map(option => {
                            return <option value={option} key={option} >{option}</option>
                        })}
                    </select>
                </div>

                    <label>
                    GROUP NAME
                    </label>
                    <input
                        type="text"
                        ref={ node => {groupname=node;}}
                        placeholder={group.group_name}
                    />


                <button type="submit">CHANGE GROUP INFO</button>
              
            </form>
        </div>
    )

}

export default ChangeGroupInfo;
