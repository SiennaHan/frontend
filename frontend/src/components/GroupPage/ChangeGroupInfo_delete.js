import React from 'react';
import {connect} from 'react-redux';
import { toChangeGroupInfo } from '../../actions';

class ChangeGroupInfo extends React.Component {
	type_options = [
		'Select an Option',
		'Club',
		'Department',
		'Else'
		]

	onSubmit = () => {
		console.log("changing group information")
		if(this.grouptype !== undefined && this.grouptype.value !== 'Select an Option' && this.groupname !== undefined) {
			this.props.onChangeGroupInfo(this.grouptype, this.groupname)
		}
	}

	render() {
		return (
			<div>
                <form onSubmit={e => {
                    e.preventDefault()
                    this.onSubmit()
                }}>
                    <div>
                        <label>
                        GROUP TYPE
                        </label>
                        <select 
                            name="grouptype"
                            ref={ node => {this.grouptype=node;} }
                            className="type-select"
                        > 
                            {this.type_options.map(option => {
                                return <option value={option} key={option} >{option}</option>
                            })}
                        </select>
                    </div>
                    <div className="Text-Field">
                        <label>
                        GROUP NAME
                        </label>
                        <input 
                            type="text" 
                            ref={ node => {this.groupname=node;}} 
                        />
                    </div>
                    <div className="Button-Field">
                    <button type="submit">CHANGE GROUP INFO</button>
                    </div>
                </form>
            </div>
		)
	}
}

const mapDispatchToProps = (dispatch) => ({
	onChangeGroupInfo: (grouptype, groupname) => dispatch(toChangeGroupInfo(grouptype, groupname))
})

export default connect(null, mapDispatchToProps)(ChangeGroupInfo);

