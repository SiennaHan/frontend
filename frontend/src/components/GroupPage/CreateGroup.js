import React from 'react'
import { connect } from 'react-redux'

import { toCreateGroup } from '../../actions/index.js'

class CreateGroup extends React.Component {
	componentDidMount() {
		this.type_options = [
			'Select an Option',
			'Club',
			'Department',
			'Else'
			]
	}
	

	onSubmit = (e) => {
		console.log("create onSubmit")
		e.preventDefault()
		if(this.grouptype !== undefined && this.grouptype.value !== 'Select an Option' && this.groupname !== undefined) {
			this.props.onCreateGroup(this.grouptype, this.groupname)
		}
	}

	render() {
		return(
			<div>
				<form onSubmit={e => {
					this.onSubmit()
				}}>


					<div className="CreateGroup">
						<label htmlFor="group type">Group Type </label>
						<select
							name="grouptype"
							ref={ node => {this.grouptype=node;} }
							className="type-select"
						>
							{this.type_options.map(option => {
								return <option value={option} key={option} >{option}</option>
							})}
						</select>
						<br />
						<br />
						<label htmlFor="group name">Group Name</label>
						<input
							type="text"
							name="groupname"
							ref={ node => {this.groupname=node;} }
							//value={this.groupname}
							className="name-input"
						/>
						<br/>
						<div className="Group-Button-Field">
						<button type="submit">CREATE GROUP</button>
						</div>
					</div>
				</form>
			</div>
		)
	}
}

const mapStateToProps = (state) => ({
	all_groups: state.all_groups,
})


const mapDispatchToProps = (dispatch) => ({
	onCreateGroup: (grouptype, groupname) => dispatch(toCreateGroup(grouptype, groupname))
})

export default connect(mapStateToProps, mapDispatchToProps)(CreateGroup)
