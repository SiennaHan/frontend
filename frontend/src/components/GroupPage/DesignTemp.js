import React from 'react';
import CommentForm from '../Comment/CommentForm';
import CommentList from '../Comment/CommentList';

const DesignTemp = ({ design, group, my_groups, onClickEdit, onClickPost, onClickLike, onClickUnlike, onClickDelete }) => {
    let post_group

    return (
        <div>
            <div>
                {console.log("design: ", design)}
                <img src={design.front_image_url} />
                <img src={design.back_image_url} />
                <p>{design.likes}</p>
            </div>
            
            {(group.group_type === "UR")
                /* grouptype이 user 그룹일때 - edit 및 post 가능 */
                ? (<div>
                        <button class="post_btn" type="button"
                            onClick={onClickEdit}>
                            EDIT
                        </button>


                        <select id="post_group" ref={node=>{post_group=node;}}>
                            <option>그룹을 선택하세요</option>
                            {my_groups.filter(group => {
                                return group.group_type !== "UR"
                            }).map(option => {
                                console.log("option: ", option)
                                return <option key={option.id} value={option.id}> {option.group_type} {option.group_name} </option>
                            })}
                        </select>

                        <button class="post_btn" type="button"
                            onClick={() => {
                                console.log("post_group: ", post_group)
                                if(post_group.value === undefined || post_group.value == "그룹을 선택하세요") {
                                    alert("그룹을 선택하세요")
                                }
                                else {
                                    onClickPost(document.getElementById("post_group").value)
                                }
                                }}>
                            POST
                        </button>
                    </div>)

                /* grouptype이 user 그룹이 아닐 때 - like 및 댓글 가능 */
                : <div>
                        {design.liked
                        ? (<button onClick={onClickUnlike}>UNLIKE DESIGN</button>)
                        : (<button onClick={onClickLike}>LIKE DESIGN</button>)
                        }
                        
                        <h2>댓글</h2>
                        {console.log("DesignTemp design.comments: ", design.comments)}
                        <CommentForm designid={design.id}/>
                        <CommentList comments={design.comments}/>
                    </div>
            }
            <br />
            {/* design의 주인일 때 - 삭제 가능 */}
            {design.auth && <button onClick={onClickDelete}>DELETE DESIGN</button>}
            <br/>
        </div>
    )
}

export default DesignTemp
