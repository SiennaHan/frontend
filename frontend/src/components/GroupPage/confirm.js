$(function() {
    $('.checked').click(function(e) {
        e.preventDefault();
        var dialog = $('<p>Are you sure?</p>').dialog({
            buttons: {
                "Yes": function() {alert('you chose yes');},
                "No":  function() {alert('you chose no');},
                "Cancel":  function() {
                    alert('you chose cancel');
                    dialog.dialog('close');
                }
            }
        });
    });
});