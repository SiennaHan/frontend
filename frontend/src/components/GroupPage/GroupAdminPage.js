import React from 'react';
import {connect} from 'react-redux';

import NavBar from '../NavBar/NavBar';
import ChangeGroupInfo from './ChangeGroupInfo';
import GroupUserList from './GroupUserList';
import GroupDesignList from './GroupDesignList';
import { toDeleteGroupUser, toDeleteGroupDesign, toChangeGroupInfo } from '../../actions';

class GroupAdminPage extends React.Component {
	render() {
		if(!this.props.loading) {
			return (
				<p>loading...</p>
			)
		}

		let admin_userlist = this.props.group_users.filter(user => {
			console.log("username: ", user.username)
			return (user.username !== this.props.user.split(":")[0])
		})

		return (
			<div className="GroupAdminPage">
				<NavBar />

				<section className="wrap clear col3">
					<div className="aside">
						<h2 className="h_white">GROUP INFO</h2>
						<div className="content">
						<ChangeGroupInfo
							group={this.props.now_group[0]}
							onClickChangeSubmit={this.props.onChangeGroupInfo}
						/>
						</div>
					</div>
					<div className="main">
						<h2 className="h_white">DESIGNS</h2>
						<div className="content">
						<GroupDesignList
							groupid={this.props.now_group[0].id}
							designlist={this.props.group_designs}
							onClickDeleteDesign={this.props.onDeleteDesign}
						/>

						</div>
					</div>
					<div className="aside">
						<h2 className="h_black">MEMBER LIST</h2>
						<div className="content">
						<GroupUserList
							groupid={this.props.now_group[0].id}
							userlist={admin_userlist}
							onClickDeleteUser={this.props.onDeleteUser}
						/>
						</div>
					</div>
				</section>
				</div>
		)
	}
}

const mapStateToProps = (state) => ({
	user: state.authorization,
	now_group: state.now_group,
    group_users: state.group_users,
    group_designs: state.group_designs,
	loading: state.loading
})

const mapDispatchToProps = (dispatch) => ({
	onChangeGroupInfo: (groupid, grouptype, groupname) => dispatch(toChangeGroupInfo(groupid, grouptype, groupname)),
	onDeleteUser: (groupid, userid) => dispatch(toDeleteGroupUser(groupid, userid)),
	onDeleteDesign: (groupid, designid) => dispatch(toDeleteGroupDesign(groupid, designid))
})

export default connect(mapStateToProps, mapDispatchToProps)(GroupAdminPage);
