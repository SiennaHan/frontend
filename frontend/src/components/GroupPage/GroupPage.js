import React from 'react';
import {connect} from 'react-redux';

import NavBar from '../NavBar/NavBar';
import CreateGroup from './CreateGroup';
import MyGroupList from './MyGroupList';
import SearchingGroup from './SearchingGroup';

class GroupPage extends React.Component {
	render() {
		if(!this.props.loading) {
			return (
				<p>loading...</p>
			)
		}

		return (
			<div>
				<NavBar />
				<CreateGroup />
				<SearchingGroup />
				<MyGroupList />
			</div>
		)
	}
}

const mapStateToProps = (state) => ({
	loading: state.loading
})

export default connect(mapStateToProps)(GroupPage);

