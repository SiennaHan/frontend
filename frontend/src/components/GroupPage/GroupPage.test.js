// import React from 'react';
// import {shallow, configure} from 'enzyme';
// import {create} from 'react-test-renderer';
// import Adapter from 'enzyme-adapter-react-15';
import * as actions from '../../actions/index'
import React from 'react';
import ReactDOM from 'react-dom';
import GroupPage from "./GroupPage";
import { SEARCH_GROUP } from '../../actions/types';
// import GroupAdminPage from "./GroupAdminPage";
// import CreateGroup from "./CreateGroup";

//configure({ adapter: new Adapter() });



describe ('ACTIONS', ()=> {
    it('should create an action to show navigation bar', () => {
        const expectedAction = {
            type: 'NAV_BAR'
        }
        expect(actions.showNavBar()).toEqual(expectedAction)
    })

    it('should create an action to sign out', () => {
        const expectedAction = {
            type: 'SIGN_OUT'

        }
        expect(actions.signOut()).toEqual(expectedAction)
    })

    it('should create an action to search group', (newList) => {
        const expectedAction = {
            type: SEARCH_GROUP,
            newList: newList

        }
        expect(actions.toSearchGroup()).toEqual(expectedAction)
    })

    it('should create an action to like design in group page', (designid) => {
        const expectedAction = {
            type: LIKE_DESIGN,
            designid: designid,

        }
        expect(actions.toLikeDesign()).toEqual(expectedAction)
    })
})

describe('GroupPage', () => {
    //ReactDOM.render(<GroupPage/>, div);
    //ReactDOM.unmountComponentAtNode(div);

    // let component = null

    // it('renders correctly', () => {
    //     component = create(<GroupPage />)
    // })

    // it('should render a CREATE GROUP h2', () => {
    //     const wrapper = shallow(
    //         <h2 className="h_white">CREATE GROUP</h2>
    //     )
    //     expect(wrapper).toMatchSnapshot()
    // })

    // it('should render CREATE GROUP component', () => {
    //     const wrapper = shallow(
    //         <CreateGroup />
    //     )
    //     expect(wrapper).toMatchSnapshot()
    // })

    // it('matches snapshot', () => {
    //     const tree = component.toJSON()
    //     expect(tree).toMatchSnapshot()
    // })

    // describe('CreateGroup', () => {
    //     // group 생성이 잘 되는지 확인
    //     it('creates group correctly', () => {

    //     })
    // })

    // describe('SearchingGroup', () => {
    //     // 검색이 잘 되는지 확인
    //     it('searches correctly', () => {

    //     })
    // })

    // describe('MyGroupList')
})


// describe('GroupAdminPage', () => {
//     let component = null

//     it('renders correctly', () => {
//         component = renderer.create(<GroupAdminPage />)
//     })
//     describe('ChangeGroupInfo')
//     describe('GroupUser')
//     describe('GroupDesign')
// })

// describe('GroupDetailPage', () => {
// })