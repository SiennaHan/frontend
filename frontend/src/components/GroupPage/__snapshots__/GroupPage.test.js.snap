// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupPage should render a CREATE GROUP h2 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <h2
    className="h_white"
>
    CREATE GROUP
</h2>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "CREATE GROUP",
      "className": "h_white",
    },
    "ref": null,
    "rendered": "CREATE GROUP",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "CREATE GROUP",
        "className": "h_white",
      },
      "ref": null,
      "rendered": "CREATE GROUP",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
